### PR TITLE
FW-5026 ADD optimize cypress tests

### PR DESCRIPTION
**Pull request by chrisfpcc**
### Description of Changes

Add cypress tests and main core suite test

### Checklist

- [ ] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable

### Reviewers Should Do The Following:

- [x] Review the changes here
- [ ] Pull the branch and test locally

### Additional Notes

N/A

### CODE REVIEW

File: cypress/e2e/core_suite_master.cy.js [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+      'be.visible',
+    )
+    cy.contains('MORE')
+    cy.contains('MORE').click()
+    cy.contains('SHARE')
+    cy.contains('SHARE').click()
+    cy.contains('Cancel').click()
+    cy.contains('MORE').click()
+    cy.contains('QR CODE')
+    cy.contains('QR CODE').click()
```
Comment (NolanVH) [inline cypress/e2e/core_suite_master.cy.js:406]: Adding the second `.contains` for clicking is redundant in this block. 
Eg this: 
```
cy.contains('MORE')
cy.contains('MORE').click()
```
could be simplified to this:
```
cy.contains('MORE').click()
```

File: cypress/e2e/core_suite_master.cy.js
Comment (NolanVH) [inline cypress/e2e/core_suite_master.cy.js:1]: This file is very large and contains an assortment of different tests. It should probably be split up into individual files organized by the component or area of the site being tested.

**Review by NolanVH — CHANGES_REQUESTED** [review]
Thanks for the large update to many of the Cypress tests @chrisfpcc!

These changes are looking great but there are a few issues I have spotted.

One of the bigger things is that some of the tests are still pointing at Nuxeo endpoints which should be removed. It also looks like the cypress.config file has the base URL hardcoded and pointing at the pre-prod site running the old Nuxeo backend. I was not able to run the tests pointed at my local environment without several code changes because of this.

The other thing that would be useful for others is some setup instructions in the README explaining what environment variables need to be set, what commands run the tests, etc. This could be done as a separate ticket.

File: cypress/e2e/dashboard/alpha.cy.js
[the remark this answers is not in the document]
Comment (chrisfpcc) [reply cypress/e2e/dashboard/alpha.cy.js:1]: this has been taken care of, removed

File: cypress/e2e/dashboard/create_page.cy.js [hunk 2]
```diff
@@ -2,74 +2,67 @@
...
+    // })
 
-  it('View Create Page - check vaidation', () => {
-    cy.visit(`${Cypress.env('baseUrl')}/` + `/nuxeo/login.jsp`)
-    cy.login(
-      Cypress.env('CYPRESS_FV_USERNAME'),
-      Cypress.env('CYPRESS_FV_PASSWORD'),
-    )
-    cy.visit(`${Cypress.env('baseUrl')}/${Cypress.env('DIALECT')}`)
+    // it('View Create Page - check vaidation', () => {
```
[the remark this answers is not in the document]
Comment (chrisfpcc) [reply cypress/e2e/dashboard/create_page.cy.js]: removed commented out areas

File: cypress/e2e/dashboard/wordup.cy.js [hunk 3]
```diff
@@ -9,13 +10,14 @@ function middlestuff() {
...
+  cy.contains('Add Categories').click()
+  cy.wait(4000)
   cy.get('.text-left > .flex-col > :nth-child(1) > :nth-child(1)').click()
   cy.contains('Add Category to document').click()
 
-  cy.contains('Add Notes').click()
+  cy.contains('Add Note').click()
   cy.get(
-    ':nth-child(6) > .mt-2 > .space-y-2 > .justify-between > .w-full',
+    ':nth-child(6) > .mt-2 > .space-y-2 > li > .justify-between > .w-full',
```
[the remark this answers is not in the document]
Comment (chrisfpcc) [reply cypress/e2e/dashboard/wordup.cy.js:20]: will make jira's to get devs to add in id's, name's

File: cypress/e2e/dashboard/pages.cy.js [hunk 4]
```diff
@@ -42,20 +52,20 @@ describe('Page Test', () => {
...
+      .eq(3)
+      .children()
+      .click()
     cy.contains('Widget').click()
     cy.contains('Alphabet').click()
     cy.contains('OK').click()
 
-    const widgets = [
-      'Mobile App',
-      'Contact Us',
```
[the remark this answers is not in the document]
Comment (chrisfpcc) [reply cypress/e2e/dashboard/pages.cy.js:54]: good to know

File: cypress/e2e/core_suite_master.cy.js [hunk 5]
```diff
@@ -0,0 +1,674 @@
...
+  cy.contains('Next Step').click()
+  cy.contains('Finish').click()
+  cy.wait(2000)
+  cy.contains('Edit').click()
+  cy.wait(1000)
+  cy.contains('Edit Words').click()
+}
+
+function _login() {
+  cy.visit(`${Cypress.env('baseUrl')}`)
```
[the remark this answers is not in the document]
Comment (chrisfpcc) [reply cypress/e2e/core_suite_master.cy.js:34]: will make a jira for it to be done in the future

File: cypress/e2e/core_suite_master.cy.js [hunk 6]
```diff
@@ -0,0 +1,674 @@
...
+    //   .trigger('mousedown', { which: 1, force: true })
+    //   .trigger('mousemove', {
+    //     clientX: 276,
+    //     clientY: 339,
+    //     screenX: 276,
+    //     screenY: 339,
+    //     pageX: 276,
+    //     pageY: 339,
+    //   })
+    //   .trigger('mouseup', { force: true, which: 1 })
```
[the remark this answers is not in the document]
Comment (chrisfpcc) [reply cypress/e2e/core_suite_master.cy.js:346]: will be removed once this solution is fixed

**Review by NolanVH — APPROVED** [review]
Some great changes here! Thanks @chrisfpcc! 

Would be great to get some documentation added to the README on how to run these tests locally (what environment variables need to be set, commands to run, etc) so that devs can run these tests before pushing changes, but that can be added in a future PR.

**Review by sahil97 — APPROVED** [review]
Looks good and +1 for Nolan's comment related to documentation.
Nice work @chrisfpcc

**Pull request merged**